runtime/src/common/sgx/mod.rs

```diff
@@ -112,3 +116,30 @@ pub struct VerifiedQuote {
     pub identity: EnclaveIdentity,
     pub timestamp: i64,
 }
+
+/// Generate a report for the given target enclave.
+#[cfg(target_env = "sgx")]
+pub fn report_for(target_info: &sgx_isa::Targetinfo, report_data: &[u8; 64]) -> sgx_isa::Report {
+    sgx_isa::Report::for_target(target_info, report_data)
+}
+
+/// Generate a report for the given target enclave.
+#[cfg(not(target_env = "sgx"))]
+pub fn report_for(_target_info: &sgx_isa::Targetinfo, report_data: &[u8; 64]) -> sgx_isa::Report {
+    let ei = EnclaveIdentity::current().unwrap();
```

Comment:
This should panic on non sgx non debug-mock-sgx builds.

Reply:
It already does as those builds will return `None`.

Reply:
What I meant was that this will panic on those builds. Is that ok or should we handle this differently?